pkg/eventprocessor/eventfeed/impl/eventfeed.go

```diff
@@ -122,6 +128,7 @@ func (ef *EventFeed) Start(
 
 	// Listen for new blocks, and get new events.
 	for h := range chHeads {
+		ef.sm.SetLastSeenBlockNumber(ef.chainID, h.Number.Int64())
```

Comment:
This is the place where the block number is being set in the shared memory object. It's right after we detected a new block header.